internal/storage/common/resource.go

```diff
@@ -100,45 +97,44 @@ func (r *ResourceRepository[ResourceType, OptionsType]) validateFilters(builder
 	}
 
 	ret := make(map[string]any)
-	properties := r.resourceHandler.Filters()
+	properties := r.resourceHandler.Schema().Fields
 	if err := builder.Walk(func(operator string, key string, value any) (err error) {
 
-		found := false
-		for _, property := range properties {
-			if len(property.Matchers) > 0 {
-				for _, matcher := range property.Matchers {
-					if found = matcher(key); found {
-						break
-					}
+		for name, property := range properties {
+			key := key
+			if property.Type.IsIndexable() {
+				key = strings.Split(key, "[")[0]
+			}
+			match := func() bool {
```

Comment:
Since this is in a for loop it might be nice to have this inline function defined somewhere else. It can be harder to track the scope of returns and breaks when there are nested for loops and functions, so moving the function definition outside would improve readability / maintainability.

Reply:
I have added a function `matchKey` to the `Field` structure.
Is it better?